grapher/stackedCharts/MarimekkoChart.tsx

```diff
@@ -0,0 +1,1655 @@
+    }
+
+    @action.bound private onEntityMouseLeave(): void {
+        this.hoveredEntityName = undefined
+    }
+
+    @action.bound private onEntityClick(entityName: string): void {
+        this.onSelectEntity(entityName)
+    }
+
+    @action.bound private onSelectEntity(entityName: string): void {
+        if (this.canAddCountry) this.selectionArray.toggleSelection(entityName)
+    }
+    @computed private get canAddCountry(): boolean {
+        const { addCountryMode } = this.manager
+        return (addCountryMode &&
+            addCountryMode !== EntitySelectionMode.Disabled) as boolean
+    }
+
+    render(): JSX.Element {
+        if (this.failMessage)
+            return (
+                <NoDataModal
+                    manager={this.manager}
+                    bounds={this.bounds}
+                    message={this.failMessage}
+                />
+            )
+
+        const { bounds, dualAxis } = this
+
+        return (
+            <g ref={this.base} className="MarimekkoChart">
+                <rect
+                    x={bounds.left}
+                    y={bounds.top}
+                    width={bounds.width}
+                    height={bounds.height}
+                    opacity={0}
+                    fill="rgba(255,255,255,0)"
+                />
+                <DualAxisComponent
+                    dualAxis={dualAxis}
+                    showTickMarks={true}
+                    horizontalAxisLabelsOnTop={true}
+                />
+                <HorizontalCategoricalColorLegend manager={this} />
+                {this.renderBars()}
+            </g>
+        )
+    }
+
+    private renderBars(): JSX.Element[] {
+        const normalElements: JSX.Element[] = []
+        const highlightedElements: JSX.Element[] = [] // highlighted elements have a thicker stroke and should be drawn last to overlap others
+        const {
+            dualAxis,
+            x0,
+            y0,
+            xDomainCorrectionFactor,
+            focusSeriesName,
+            placedLabels,
+            labelLines,
+            placedItems,
+            hoveredEntityName,
+            mustEnsureOnePixelXSize,
+        } = this
+        const selectionSet = this.selectionArray.selectedSet
+        const targetTime = this.manager.endTime
+        const timeColumn = this.inputTable.timeColumn
+        const yAxisColumn = this.formatColumn
+        const xAxisColumn = this.xColumn
+        const labelYOffset = 0
+        const hasSelection = selectionSet.size > 0
+        let noDataAreaElement = undefined
+        let noDataLabel = undefined
+        let patterns: JSX.Element[] = []
+        const noDataHeight = Bounds.forText("no data").height + 10 //  dualAxis.verticalAxis.rangeSize
+
+        const firstNanValue = placedItems.findIndex((item) => !item.bars.length)
+        const anyNonNanAfterFirstNan =
+            firstNanValue >= 0
+                ? drop(placedItems, firstNanValue).some(
```

Comment:
nit: could use the native `.slice()` here, I think `drop(xs, n)` === `xs.slice(n)`